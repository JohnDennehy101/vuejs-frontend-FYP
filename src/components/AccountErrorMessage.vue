<template>
  <div class="form-control" v-show="toggle" id="error-sign-up-notice">
    <i class="fas fa-exclamation-triangle"></i>
    <p>{{ errorMessage }}</p>

    <span @click="hideErrorMessage">
      <i class="fas fa-times-circle"></i>
    </span>
  </div>
</template>

<script>
export default {
  props: {
    toggle: {
      type: Boolean,
    },
    errorMessage: {
      type: String,
    },
  },
  data() {
    return {
      display: this.toggle,
    };
  },

  methods: {
    hideErrorMessage() {
      this.$emit("hideErrorMessage", true);
    },
  },
};
</script>

<style scoped lang="scss">
#error-sign-up-notice {
  border-radius: 5px;
  background-color: #e57373;
  color: white;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0.5rem;

  p {
    margin-left: 0.5rem;
    margin-right: 0.5rem;
  }
  i {
    pointer-events: all;
  }
  .form-control {
    width: 80%;
    margin: 0.5rem auto;
  }
}
</style>
