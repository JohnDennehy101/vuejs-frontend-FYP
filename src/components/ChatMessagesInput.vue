<template>
  <div id="add-chat-message-container">
    <div id="new-chat-message-components">
      <h4>Add Comment</h4>
      <textarea
        id="message"
        name="message"
        type="text"
        placeholder="Type your comment here"
        v-model="message"
      />

      <button
        className="hide-event-detail-category-button"
        v-on:click="addComment"
      >
        Post Comment
      </button>
    </div>
  </div>
</template>

<script>
export default {
  data() {
    return {
      message: "",
    };
  },
  methods: {
    addComment(event) {
      event.preventDefault();

      this.$emit("addComment", this.message);
      this.message = "";
    },
  },
};
</script>

<style scoped lang="scss">
#add-chat-message-container {
  width: 80%;

  margin: 1rem auto;
  border-radius: 5px;
  background-color: #ffffff;

  @include for-phone-only {
    width: 90%;
  }
}
#new-chat-message-components {
  padding: 1.5rem;

  @include for-phone-only {
    padding: 1.5rem 1rem;
  }
}
textarea {
  display: block;
  width: 98%;
  height: 15vh;
  border: none;
  background-color: #f7f8fd;
  margin: 1rem auto;
  padding-left: 1rem;
  padding-top: 1rem;
  font-family: "Open Sans", sans-serif;
  resize: none;

  @include for-phone-only {
    width: 90%;
  }
}
h4 {
  margin-left: 0.5rem;
}

button {
  background-color: #3a4374;
  color: #ffffff;
  border-radius: 10px;
  border: none;
  width: 20%;
  height: 2.5rem;
  margin-left: 5px;

  @include for-phone-only {
    width: 40%;
    height: 2rem;
  }
}
button:hover {
  cursor: pointer;
}
</style>
