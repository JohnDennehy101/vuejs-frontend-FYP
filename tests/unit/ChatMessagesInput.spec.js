import { mount, shallowMount } from "@vue/test-utils";
import ChatMessagesInput from "@/components/ChatMessagesInput.vue";
import AccountErrorMessage from "@/components/AccountErrorMessage";
import mockSuccessfulUserService from "./mocks/userService.mock";
import mockStore from "./mocks/mockStore.mock";
import mockRouter from "./mocks/mockRouter.mock";
import { nextTick } from "vue";

const mockRoute = {
  params: {
    id: 1,
  },
};

describe("ChatMessagesInput.vue", () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  it("should render correctly", () => {
    const wrapper = shallowMount(ChatMessagesInput);
    expect(wrapper).toMatchSnapshot();
  });
  it("message added on click of Post comment button", async () => {
    const wrapper = mount(ChatMessagesInput);

    wrapper.find("#message").setValue("Test Message");

    await wrapper.find(".hide-event-detail-category-button").trigger("click");
    expect(wrapper.emitted()).toHaveProperty("addComment");
  });
});
