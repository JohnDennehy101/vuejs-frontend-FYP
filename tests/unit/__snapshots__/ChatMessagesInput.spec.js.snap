// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatMessagesInput.vue should render correctly 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "name": "VTU_ROOT",
      "render": [Function],
      "setup": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <div
        id="add-chat-message-container"
      >
        <div
          id="new-chat-message-components"
        >
          <h4>
            Add Comment
          </h4>
          <textarea
            id="message"
            name="message"
            placeholder="Type your comment here"
            type="text"
          />
          <button
            class="hide-event-detail-category-button"
          >
             Post Comment 
          </button>
        </div>
      </div>
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {
        "transition": Object {
          "name": "transition",
        },
        "transition-group": Object {
          "name": "transition-group",
        },
      },
      "config": Object {
        "compilerOptions": Object {},
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "emitsCache": WeakMap {},
      "mixins": Array [
        Object {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": Object {},
      "reload": [Function],
    },
    "_instance": Object {
      "a": null,
      "accessCache": Object {},
      "appContext": Object {
        "app": [Circular],
        "components": Object {
          "transition": Object {
            "name": "transition",
          },
          "transition-group": Object {
            "name": "transition-group",
          },
        },
        "config": Object {
          "compilerOptions": Object {},
          "errorHandler": undefined,
          "globalProperties": Object {},
          "isNativeTag": [Function],
          "optionMergeStrategies": Object {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": Object {},
        "emitsCache": WeakMap {},
        "mixins": Array [
          Object {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": Object {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": Object {},
      "attrsProxy": null,
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": Object {
        "VTU_COMPONENT": Object {
          "addComment": [Function],
          "message": "",
        },
      },
      "da": null,
      "data": Object {},
      "devtoolsRawSetupState": Object {
        "VTU_COMPONENT": RefImpl {
          "__v_isRef": true,
          "__v_isShallow": false,
          "_rawValue": Object {
            "addComment": [Function],
            "message": "",
          },
          "_value": Object {
            "addComment": [Function],
            "message": "",
          },
          "dep": undefined,
        },
      },
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "active": true,
        "allowRecurse": true,
        "deps": Array [
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "parent": undefined,
        "scheduler": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": Object {},
      "propsDefaults": Object {},
      "propsOptions": Array [
        Object {},
        Array [],
      ],
      "provides": Object {},
      "proxy": Object {
        "VTU_COMPONENT": Object {
          "addComment": [Function],
          "message": "",
        },
      },
      "refs": Object {
        "VTU_COMPONENT": Object {
          "addComment": [Function],
          "message": "",
        },
      },
      "render": [Function],
      "renderCache": Array [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "_active": true,
        "cleanups": Array [],
        "detached": true,
        "effects": Array [
          ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": Array [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
        ],
        "parent": undefined,
      },
      "setupContext": null,
      "setupState": Object {
        "VTU_COMPONENT": Object {
          "addComment": [Function],
          "message": "",
        },
      },
      "slots": Object {},
      "slotsProxy": null,
      "sp": null,
      "subTree": Object {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": Object {
          "a": null,
          "accessCache": Object {
            "@@__IMMUTABLE_ITERABLE__@@": 0,
            "@@__IMMUTABLE_RECORD__@@": 0,
            "__v_isReadonly": 0,
            "__v_isRef": 0,
            "__v_isShallow": 0,
            "__v_raw": 0,
            "__v_skip": 4,
            "_isMockFunction": 0,
            "addComment": 4,
            "constructor": 0,
            "isVueInstance": 0,
            "message": 2,
            "toJSON": 0,
            Symbol(Symbol.toStringTag): 0,
          },
          "appContext": Object {
            "app": [Circular],
            "components": Object {
              "transition": Object {
                "name": "transition",
              },
              "transition-group": Object {
                "name": "transition-group",
              },
            },
            "config": Object {
              "compilerOptions": Object {},
              "errorHandler": undefined,
              "globalProperties": Object {},
              "isNativeTag": [Function],
              "optionMergeStrategies": Object {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": Object {},
            "emitsCache": WeakMap {},
            "mixins": Array [
              Object {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": Object {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": Object {},
          "attrsProxy": null,
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": Object {},
          "ctx": Object {
            "addComment": [Function],
            "message": "",
          },
          "da": null,
          "data": Object {
            "message": "",
          },
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": Array [
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
          "emit": [Function],
          "emitsOptions": null,
          "emitted": null,
          "exposeProxy": null,
          "exposed": null,
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": Object {},
          "propsDefaults": Object {},
          "propsOptions": Array [
            Object {},
            Array [],
          ],
          "provides": Object {},
          "proxy": Object {
            "addComment": [Function],
            "message": "",
          },
          "refs": Object {},
          "render": [Function],
          "renderCache": Array [
            [Function],
            [Function],
          ],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "_active": true,
            "cleanups": Array [],
            "detached": true,
            "effects": Array [
              ReactiveEffect {
                "active": true,
                "allowRecurse": true,
                "deps": Array [
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "parent": undefined,
                "scheduler": [Function],
              },
            ],
            "parent": undefined,
          },
          "setupContext": null,
          "setupState": Object {},
          "slots": Object {},
          "slotsProxy": null,
          "sp": null,
          "subTree": Object {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": null,
            "appContext": null,
            "children": Array [
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": Array [
                  Object {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "ce": undefined,
                    "children": "Add Comment",
                    "component": null,
                    "ctx": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <h4>
                      Add Comment
                    </h4>,
                    "key": null,
                    "patchFlag": -1,
                    "props": null,
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 9,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "h4",
                  },
                  Object {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": null,
                    "component": null,
                    "ctx": [Circular],
                    "dirs": Array [
                      Object {
                        "arg": undefined,
                        "dir": Object {
                          "beforeUpdate": [Function],
                          "created": [Function],
                          "getSSRProps": [Function],
                          "mounted": [Function],
                        },
                        "instance": Object {
                          "addComment": [Function],
                          "message": "",
                        },
                        "modifiers": Object {},
                        "oldValue": undefined,
                        "value": "",
                      },
                    ],
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <textarea
                      id="message"
                      name="message"
                      placeholder="Type your comment here"
                      type="text"
                    />,
                    "key": null,
                    "patchFlag": 512,
                    "props": Object {
                      "id": "message",
                      "name": "message",
                      "onUpdate:modelValue": [Function],
                      "placeholder": "Type your comment here",
                      "type": "text",
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 1,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "textarea",
                  },
                  Object {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": " Post Comment ",
                    "component": null,
                    "ctx": [Circular],
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <button
                      class="hide-event-detail-category-button"
                    >
                       Post Comment 
                    </button>,
                    "key": null,
                    "patchFlag": 0,
                    "props": Object {
                      "className": "hide-event-detail-category-button",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 9,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "button",
                  },
                ],
                "component": null,
                "ctx": [Circular],
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <div
                  id="new-chat-message-components"
                >
                  <h4>
                    Add Comment
                  </h4>
                  <textarea
                    id="message"
                    name="message"
                    placeholder="Type your comment here"
                    type="text"
                  />
                  <button
                    class="hide-event-detail-category-button"
                  >
                     Post Comment 
                  </button>
                </div>,
                "key": null,
                "patchFlag": 0,
                "props": Object {
                  "id": "new-chat-message-components",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 17,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "div",
              },
            ],
            "component": null,
            "ctx": [Circular],
            "dirs": null,
            "dynamicChildren": Array [
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": null,
                "component": null,
                "ctx": [Circular],
                "dirs": Array [
                  Object {
                    "arg": undefined,
                    "dir": Object {
                      "beforeUpdate": [Function],
                      "created": [Function],
                      "getSSRProps": [Function],
                      "mounted": [Function],
                    },
                    "instance": Object {
                      "addComment": [Function],
                      "message": "",
                    },
                    "modifiers": Object {},
                    "oldValue": undefined,
                    "value": "",
                  },
                ],
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <textarea
                  id="message"
                  name="message"
                  placeholder="Type your comment here"
                  type="text"
                />,
                "key": null,
                "patchFlag": 512,
                "props": Object {
                  "id": "message",
                  "name": "message",
                  "onUpdate:modelValue": [Function],
                  "placeholder": "Type your comment here",
                  "type": "text",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "textarea",
              },
            ],
            "dynamicProps": null,
            "el": <div
              id="add-chat-message-container"
            >
              <div
                id="new-chat-message-components"
              >
                <h4>
                  Add Comment
                </h4>
                <textarea
                  id="message"
                  name="message"
                  placeholder="Type your comment here"
                  type="text"
                />
                <button
                  class="hide-event-detail-category-button"
                >
                   Post Comment 
                </button>
              </div>
            </div>,
            "key": null,
            "patchFlag": 0,
            "props": Object {
              "id": "add-chat-message-container",
            },
            "ref": null,
            "scopeId": null,
            "shapeFlag": 17,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "transition": null,
            "type": "div",
          },
          "suspense": null,
          "suspenseId": 0,
          "type": Object {
            "components": Object {},
            "data": [Function],
            "methods": Object {
              "addComment": [Function],
            },
            "render": [Function],
          },
          "u": null,
          "uid": 1,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "ctx": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          id="add-chat-message-container"
        >
          <div
            id="new-chat-message-components"
          >
            <h4>
              Add Comment
            </h4>
            <textarea
              id="message"
              name="message"
              placeholder="Type your comment here"
              type="text"
            />
            <button
              class="hide-event-detail-category-button"
            >
               Post Comment 
            </button>
          </div>
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": Object {
          "ref": "VTU_COMPONENT",
        },
        "ref": Object {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": Object {
          "components": Object {},
          "data": [Function],
          "methods": Object {
            "addComment": [Function],
          },
          "render": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": Object {
        "name": "VTU_ROOT",
        "render": [Function],
        "setup": [Function],
      },
      "u": null,
      "uid": 0,
      "um": null,
      "update": [Function],
      "vnode": Object {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": Object {
          "app": [Circular],
          "components": Object {
            "transition": Object {
              "name": "transition",
            },
            "transition-group": Object {
              "name": "transition-group",
            },
          },
          "config": Object {
            "compilerOptions": Object {},
            "errorHandler": undefined,
            "globalProperties": Object {},
            "isNativeTag": [Function],
            "optionMergeStrategies": Object {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": Object {},
          "emitsCache": WeakMap {},
          "mixins": Array [
            Object {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": Object {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "ctx": null,
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          id="add-chat-message-container"
        >
          <div
            id="new-chat-message-components"
          >
            <h4>
              Add Comment
            </h4>
            <textarea
              id="message"
              name="message"
              placeholder="Type your comment here"
              type="text"
            />
            <button
              class="hide-event-detail-category-button"
            >
               Post Comment 
            </button>
          </div>
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": Object {
          "name": "VTU_ROOT",
          "render": [Function],
          "setup": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": Object {
      "compilerOptions": Object {},
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "runWithContext": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.3.4",
  },
  "__setProps": [Function],
  "cleanUpCallbacks": Array [
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
  ],
  "componentVM": Object {
    "addComment": [Function],
    "message": "",
  },
  "isDisabled": [Function],
  "rootVM": Object {
    "VTU_COMPONENT": Object {
      "addComment": [Function],
      "message": "",
    },
  },
  "wrapperElement": <div
    id="add-chat-message-container"
  >
    <div
      id="new-chat-message-components"
    >
      <h4>
        Add Comment
      </h4>
      <textarea
        id="message"
        name="message"
        placeholder="Type your comment here"
        type="text"
      />
      <button
        class="hide-event-detail-category-button"
      >
         Post Comment 
      </button>
    </div>
  </div>,
}
`;
